src/android/LocalNotification.java

```diff
@@ -132,57 +158,228 @@ public boolean execute (final String action, final JSONArray args,
                     clearAll(command);
-                } else
-                if (action.equals("type")) {
+                } else if (action.equals("type")) {
                     type(args, command);
-                } else
-                if (action.equals("ids")) {
+                } else if (action.equals("ids")) {
                     ids(args, command);
-                } else
-                if (action.equals("notification")) {
+                } else if (action.equals("notification")) {
                     notification(args, command);
-                } else
-                if (action.equals("notifications")) {
+                } else if (action.equals("notifications")) {
                     notifications(args, command);
+                } else if (action.equals("hasDoNotDisturbPermissions")) {
+                    hasDoNotDisturbPermissions(command);
+                } else if (action.equals("requestDoNotDisturbPermissions")) {
+                    requestDoNotDisturbPermissions(command);
+                } else if (action.equals("isIgnoringBatteryOptimizations")) {
+                    isIgnoringBatteryOptimizations(command);
+                } else if (action.equals("requestIgnoreBatteryOptimizations")) {
+                    requestIgnoreBatteryOptimizations(command);
+                } else if (action.equals("dummyNotifications")) {
+                    dummyNotifications(command);
                 }
             }
         });
 
         return true;
     }
 
+    /**
+     * Required for Android 13 to get the runtime notification permissions.
+     *
+     * @param command The callback context used when calling back into JavaScript.
+     */
+    private void dummyNotifications(CallbackContext command) {
+
+        fireEvent("dummyNotifications");
+        NotificationManager mNotificationManager;
+        NotificationCompat.Builder mBuilder;
+        String NOTIFICATION_CHANNEL_ID = "10004457";
+        String notificationMsg = "Test";
+        String notificationTitle = "Mdd";
+        Context context =  cordova.getActivity().getApplicationContext();
+
+        Intent intentToLaunch = new Intent(context, TriggerReceiver.class);
+        intentToLaunch.putExtra("Callfrom", "reminders");
+
+        final PendingIntent resultPendingIntent = PendingIntent.getActivity(context,
+                0, intentToLaunch, PendingIntent.FLAG_IMMUTABLE);
+
+        mBuilder = new NotificationCompat.Builder(context,NOTIFICATION_CHANNEL_ID);
+        mBuilder.setContentIntent(resultPendingIntent);
+
+        mNotificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O){
+            int importance = NotificationManager.IMPORTANCE_HIGH;
+            NotificationChannel notificationChannel = new NotificationChannel(NOTIFICATION_CHANNEL_ID, "NOTIFICATION_CHANNEL_NAME", importance);
```

Comment:
This is creating a Channel with the name "NOTIFICATION_CHANNEL_NAME" in Android >= 8. The channel name should be configurable or should use the same logic as creating a normal notification.

Reply:
The Method `Options.getChannel` is doing, what we need.

Reply:
should be fixed now, because I've added a new argument for 'options' (merged with plugin default options)